### PR TITLE
Pool 2870 update prizepool vault to be able to redistribute prizes

### PR DESCRIPTION
**Pull request by asselstine**

### CODE REVIEW

**Comment by asselstine** [comment]
Note: compilation was having trouble so I also removed the dependency on the claimer

File: src/interfaces/IVaultHooks.sol [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+pragma solidity 0.8.17;
+
+struct Hook {
+    /// @notice If true, the vault will call the beforeClaimPrize fxn on the hook.
+    bool useBeforeClaimPrize;
+    /// @notice If true, the vault will call the afterClaimPrize on the hook
+    bool useAfterClaimPrize;
+    /// @notice The address of the smart contarct implementing the hooks
+    IVaultHooks hooks;
+}
```
Comment (trmid) [inline src/interfaces/IVaultHooks.sol:11]: It's odd having `Hook.hooks` as a field. We could do something like:
```suggestion
struct VaultHooks {
    /// @notice If true, the vault will call the beforeClaimPrize fxn on the hook.
    bool useBeforeClaimPrize;
    /// @notice If true, the vault will call the afterClaimPrize on the hook
    bool useAfterClaimPrize;
    /// @notice The address of the smart contract implementing the hooks
    IVaultHooks implementation;
}
```

File: src/Vault.sol [hunk 2]
```diff
@@ -569,20 +579,47 @@ contract Vault is ERC4626, ERC20Permit, ILiquidationSource, Ownable {
...
+      recipient = hook.hooks.beforeClaimPrize(_winner, _tier, _prizeIndex);
+    } else {
+      recipient = _winner;
+    }
+
+    uint prizeTotal = _prizePool.claimPrize(_winner, _tier, _prizeIndex, recipient, _fee, _feeRecipient);
+
+    if (hook.useAfterClaimPrize) {
+      hook.hooks.afterClaimPrize(_winner, _tier, _prizeIndex, prizeTotal - _fee, recipient);
+    }
```
Comment (trmid) [inline src/Vault.sol:620]: `hook.hooks` is a bit odd. We could change it to:
```suggestion
    VaultHooks memory hooks = _hooks[_winner];
    address recipient;
    if (hooks.useBeforeClaimPrize) {
      recipient = hooks.implementation.beforeClaimPrize(_winner, _tier, _prizeIndex);
    } else {
      recipient = _winner;
    }

    uint prizeTotal = _prizePool.claimPrize(_winner, _tier, _prizeIndex, recipient, _fee, _feeRecipient);

    if (hooks.useAfterClaimPrize) {
      hooks.implementation.afterClaimPrize(_winner, _tier, _prizeIndex, prizeTotal - _fee, recipient);
    }
```

**Review by trmid — CHANGES_REQUESTED** [review]
Added some suggestions for name changes on the hook struct to make it less repetitive.

**Review by trmid — APPROVED** [review]

**Pull request merged**